beacon_node/http_api/src/attestation_rewards.rs

```diff
@@ -0,0 +1,260 @@
+            .map_err(|_| {
+                warp_utils::reject::custom_server_error("Unable to get eligible".to_owned())
+            })?;
+
+        let balance = state.get_balance_mut(*validator_index).map_err(|_| {
+            warp_utils::reject::custom_server_error("Unable to get balance".to_owned())
+        })?;
+        *balance = (*balance as f64).round().clamp(0.0, 32.0) as u64;
+
+        let mut head_reward = 0u64;
+        let mut target_reward = 0u64;
+        let mut source_reward = 0u64;
+
+        for &flag_index in [
+            TIMELY_SOURCE_FLAG_INDEX,
+            TIMELY_TARGET_FLAG_INDEX,
+            TIMELY_HEAD_FLAG_INDEX,
+        ]
+        .iter()
+        {
+            if eligible {
+                let voted_correctly = participation_cache
+                    .get_unslashed_participating_indices(flag_index, previous_epoch)
+                    .is_ok();
+                if voted_correctly {
+                    let _ideal_reward = &ideal_rewards
+                        .iter()
+                        .find(|reward| reward.effective_balance == *balance)
+                        .map(|reward| {
+                            head_reward = reward.head;
+                            target_reward = reward.target;
+                            source_reward = reward.source;
+                            reward
+                        })
+                        .unwrap_or(&IdealAttestationRewards {
+                            effective_balance: *balance,
+                            head: 0,
+                            target: 0,
+                            source: 0,
+                        });
+                } else {
+                    match flag_index {
+                        TIMELY_HEAD_FLAG_INDEX => {}
+                        TIMELY_TARGET_FLAG_INDEX => {
+                            target_reward = (-(base_reward as i64 as i128) * weight as i128
+                                / WEIGHT_DENOMINATOR as i128)
+                                as u64
+                        }
+                        TIMELY_SOURCE_FLAG_INDEX => {
+                            source_reward = (-(base_reward as i64 as i128) * weight as i128
+                                / WEIGHT_DENOMINATOR as i128)
+                                as u64
+                        }
+                        _ => {}
+                    }
+                }
+            }
+
+            total_rewards.push(TotalAttestationRewards {
+                validator_index: *validator_index as u64,
+                head: head_reward as i64,
+                target: target_reward as i64,
+                source: source_reward as i64,
+                inclusion_delay: 0,
+            });
+        }
+    }
+
+    //TODO Check inclusion delay. Where do I get attestation data from?
+    //let att_data = attestations.attestation_data();
+    //let inclusion_delay = state.slot().as_u64().checked_sub(att_data.slot.as_u64())?;
```

Comment:
Ah crap, you're right. Getting access to the inclusion delay is a bit hard because we don't know from the `state` which block the attestation was included in.

I'm tempted to say we should leave this blank for now and come back to it. I think the best strategy might be to load all the blocks from the last two epochs into memory and build an index mapping validator indices to the slot where their attestation was first included. It'll be a bit of a pain.

In the case where the head reward is matched we'll know the inclusion delay is zero, and could bypass the more expensive lookup in that case.